analytics-kotlin/src/main/java/com/segment/analytics/Settings.kt

```diff
@@ -17,25 +16,42 @@ data class Settings(
     var edgeFunction: JsonObject = emptyJsonObject,
 )
 
+internal fun Analytics.update(settings: Settings) {
+    timeline.applyClosure { plugin ->
+        // tell all top level plugins to update.
+        // For destination plugins they auto-handle propagation to sub-plugins
+        plugin.update(settings)
```

Comment:
Do we care to filter destination plugins or can all plugins have "started" state?

Reply:
i think all should have it, filtering the `started` state to only destinations may make things difficult to understand in the long term